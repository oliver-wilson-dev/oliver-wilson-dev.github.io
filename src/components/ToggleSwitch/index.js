import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './index.css';

const inputId = 'input-id';

const ToggleSwitch = ({ toggleTheme, checkBoxChecked }) => (
  <label className={styles.switch} htmlFor={inputId}>
    <input type="checkbox" id={inputId} onClick={toggleTheme} defaultChecked={checkBoxChecked} />
    <span className={classnames(styles.slider, styles.round)} />
  </label>
);

ToggleSwitch.propTypes = {
  checkBoxChecked: PropTypes.bool.isRequired,
  toggleTheme: PropTypes.func.isRequired
};

export default ToggleSwitch;
