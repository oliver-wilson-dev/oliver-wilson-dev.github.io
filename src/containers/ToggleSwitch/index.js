import { connect } from 'react-redux';
import ToggleSwitch from '../../components/ToggleSwitch';
import toggleTheme from '../../state/actions/toggleTheme';

const mapDispatchToProps = dispatch => ({
  toggleTheme: () => dispatch(toggleTheme())
});

export default connect(undefined, mapDispatchToProps)(ToggleSwitch);
