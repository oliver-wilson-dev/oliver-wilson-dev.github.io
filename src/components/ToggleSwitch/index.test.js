import React from 'react';
import { shallow } from 'enzyme';
import ToggleSwitch from './index';


const defaultProps = {
  toggleTheme: jest.fn(),
  checkBoxChecked: false
};
const render = overrideProps => shallow(<ToggleSwitch {...defaultProps} {...overrideProps} />);

describe('ToggleSwitch', () => {
  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  it("should have a label with the htmlFor property that has the same value as the input element's id", () => {
    const component = render();
    expect(component.find('label').prop('htmlFor')).toBe(
      component.find('input').prop('id')
    );
  });

  it('should call the toggleTheme prop when the input is clicked', () => {
    const mockToggleTheme = jest.fn();
    const component = render({ toggleTheme: mockToggleTheme });
    component.find('input').simulate('click');

    expect(mockToggleTheme).toHaveBeenCalledTimes(1);
  });

  describe('when the theme had been toggled from a previous browse to the website', () => {
    it("should set the default click value to the value of the prop 'checkBoxChecked' prop", () => {
      expect(render({ checkBoxChecked: true })).toMatchSnapshot();
    });
  });
});
