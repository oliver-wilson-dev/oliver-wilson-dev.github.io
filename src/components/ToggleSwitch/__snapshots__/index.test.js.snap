// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch should render correctly 1`] = `
<label
  className="switch"
  htmlFor="input-id"
>
  <input
    id="input-id"
    type="checkbox"
  />
  <span
    className="slider round"
  />
</label>
`;
